### DISCUSSION TITLE
Modify the document title when the locale changes

### DISCUSSION THREAD

**Discussion by gefey397831** [intlify/vue-i18n#1049] [Q&A]
Thank you for this great project!

I'm using Vue 3 + Vite + Vue I18n 9.

I want to modify the `<title>` and `<html lang="">` in `index.html` when the locale changes.

I used `{{ $t("message.title") }}` but it doesn't work (it works in `.vue` files). `{{ $t("message.title") }}` was rendered directly. What should I do? Thank you!

**Comment by kazupon** [comment]
You can realize by using `@vueuse/head`
https://github.com/vueuse/head

**Reply by gefey397831** [reply]
Thanks for your reply!

There are two issues:

1. `{{ $t("message.hello") }}` can only be used in the template, but `useHead` is only in `<script>`, not `<template>`;
2. There is still no way to change the `lang` of `<html>`.

**Reply by parallels999** [reply]
is this **answered**?

**Comment by it9gamelog** [comment]
Found this discussion, having the same problem.

Now my solution is https://stackoverflow.com/questions/36612847/how-can-i-bind-the-html-title-content-in-vuejs, by creating a component I could bind the t('Title') to the props.

**Reply by it9gamelog** [reply]
Or by Pinia

Store:
```
import { defineStore } from 'pinia'
import Cookies from 'js-cookie'

export const useAppStore = defineStore('app', {
  state: () => ({
    title: null
  })
})

```

Child Page:
```
<script setup>
import { watchEffect } from 'vue'
import { useI18n } from 'vue-i18n'
import { useAppStore } from 'stores/app'

const { t } = useI18n()

const appStore = useAppStore()
watchEffect(() => {
  appStore.title = t('title')
})
</script>
```

Index/Layout Page:
```
<script setup>
import { useAppStore } from 'stores/app'
const appStore = useAppStore()

watch(() => appStore.title, () => {
  document.title = appStore.title
})
</scirpt>
```
